src/python/include/Persistent_cohomology_interface.h

```diff
@@ -73,6 +72,54 @@ persistent_cohomology::Persistent_cohomology<FilteredComplex, persistent_cohomol
     return persistence;
   }
 
+  int top_dimensional_coface(int splx){
+    if (stptr_->dimension(splx) == stptr_->dimension()){return splx;}
+    else{  
+      for (auto v : stptr_->coboundary_simplex_range(splx)){
+        if(stptr_->filtration(v) == stptr_->filtration(splx)){
+          return top_dimensional_coface(v);
+        }
+      }  
+    }
+  }
+
+  std::vector<std::vector<int>> cofaces_of_cubical_persistence_pairs() {
+
+    // Warning: this function is meant to be used with CubicalComplex only!!
+
+    auto pairs = persistent_cohomology::Persistent_cohomology<FilteredComplex,
+      persistent_cohomology::Field_Zp>::get_persistent_pairs();
+
+    // Gather all top-dimensional cells and store their simplex handles
+    std::vector<int> max_splx; for (auto splx : stptr_->top_dimensional_cells_range()){
+      max_splx.push_back(splx); 
+    }
+    // Sort these simplex handles and compute the ordering function
+    // This function allows to go directly from the simplex handle to the position of the corresponding top-dimensional cell in the input data 
+    std::map<int, int> order; std::sort(max_splx.begin(), max_splx.end()); 
+    for (unsigned int i = 0; i < max_splx.size(); i++)  order.insert(std::make_pair(max_splx[i], i));
+
+    std::vector<std::vector<int>> persistence_pairs;
+    for (auto pair : pairs) {
+      int h = stptr_->dimension(get<0>(pair));
+      // Recursively get the top-dimensional cell / coface associated to the persistence generator
+      int face0 = top_dimensional_coface(get<0>(pair));
+      // Retrieve the index of the corresponding top-dimensional cell in the input data
+      int splx0 = order[face0];
+
+      int splx1 = -1;
```

Comment:
I don't remember if it is documented somewhere that -1 means we have an unmatched simplex.

Reply:
Indeed. I updated the doc.